File: C++/3217. Delete Nodes From Linked List Present in Array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& n, ListNode* head) {
        if (!head || n.empty()) 
        return head;
        unordered_set<int>a(n.begin(),n.end());
        while (head && a.count(head->val)) { 
                head = head->next;
            }
            if(head==NULL)
            return head;
            ListNode* temp=head;
        while(temp && temp->next){
            if(a.count(temp->next->val)){
               temp->next=temp->next->next;
            }
            else
            temp=temp->next;
        }
    return head;
    }
};

File: JAVA/3217. Delete Nodes From Linked List Present in Array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] n, ListNode head) {
        if(head==null||n.length==0)
        return head;
        Set<Integer>a=new HashSet<>();
        for (int x : n) {
            a.add(x);
        }
        while(head!=null && a.contains(head.val)){
            head=head.next;
        }
        if(head==null)
        return head;
        ListNode temp=head;
        while(temp!=null &&temp.next!=null){
            if(a.contains(temp.next.val))
            temp.next=temp.next.next;
            else
            temp=temp.next;
        }
        return head;
    }
}